sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -1070,23 +1069,27 @@ export class MessageSession extends LinkEntity {
       /**
        * Resets the timer when a new message is received. If no messages were received for
        * `newMessageWaitTimeoutInSeconds`, the messages received till now are returned. The
-       * receiver link stays open for the next receive call, but doesnt receive messages until
+       * receiver link stays open for the next receive call, but doesnt receive messages until then.
+       * The new message wait timer mechanism is used only in `peekLock` mode.
        */
-      const resetTimerOnNewMessageReceived = (): void => {
-        if (this._newMessageReceivedTimer) clearTimeout(this._newMessageReceivedTimer);
-        if (this.newMessageWaitTimeoutInSeconds) {
-          this._newMessageReceivedTimer = setTimeout(async () => {
-            const msg =
-              `MessageSession '${this.sessionId}' with name '${this.name}' did not receive ` +
-              `any messages in the last ${this.newMessageWaitTimeoutInSeconds} seconds. Hence closing it.`;
-            log.error("[%s] %s", this._context.namespace.connectionId, msg);
-            finalAction();
-            if (this.callee === SessionCallee.sessionManager) {
-              await this.close();
+      const resetTimerOnNewMessageReceived =
+        this.receiveMode === ReceiveMode.peekLock
+          ? (): void => {
+              if (this._newMessageReceivedTimer) clearTimeout(this._newMessageReceivedTimer);
+              if (this.newMessageWaitTimeoutInSeconds) {
+                this._newMessageReceivedTimer = setTimeout(async () => {
+                  const msg =
+                    `MessageSession '${this.sessionId}' with name '${this.name}' did not receive ` +
+                    `any messages in the last ${this.newMessageWaitTimeoutInSeconds} seconds. Hence closing it.`;
+                  log.error("[%s] %s", this._context.namespace.connectionId, msg);
+                  finalAction();
+                  if (this.callee === SessionCallee.sessionManager) {
+                    await this.close();
+                  }
+                }, this.newMessageWaitTimeoutInSeconds * 1000);
+              }
             }
-          }, this.newMessageWaitTimeoutInSeconds * 1000);
-        }
-      };
+          : () => {};
```

Comment:
I would not suggest the use of ternary operator in this scenario as it makes the code less readable.

Since the timer is only set if `this.newMessageWaitTimeoutInSeconds` is truthy, how about setting `this.newMessageWaitTimeoutInSeconds` to 0 when receive mode is ReceiveAndDelete?

That way, the change is restricted to a smaller area and `resetTimerOnNewMessageReceived` need not be touched at all. Same applies to the non session case in batchingReceiver.ts file.

Reply:
Sure, that could be another way to go about it - given that we still have #6270 as related - trying this new approach will need to be tested again.
Otherwise, we are just adding an if else check - the diffs on PR appears to be 10 lines but it that's cause of indentation and is really just 2-3 lines - so imo, there isn't much benefit in proposed changes vs current

Furthermore current changes seem a little more direct in that we don't renew message timer for receiveAndDelete mode.
But in proposed changes, we set `this.newMessageWaitTimeoutInSeconds` and it could be changed elsewhere as well, so relying on it's value seems a little too indirect.

Reply:
So I have run 3-4 test runs and I think we should stick with current changes

Test results on current changes:
```

  Perf issues
Expected 10000 but received 10000
Expected to remove 10000 but removed 10000
    ✓ #4140 - HANGING (11576ms)


  1 passing (12s)
```

```
  Perf issues
Expected 10000 but received 10000
Expected to remove 10000 but removed 10000
    ✓ #4140 - HANGING (10967ms)


  1 passing (11s)
```

Test results on making changes where we set the `this.newMessageWaitTimeoutInSeconds` to 0

```

Received transfer when credit was 0
Expected 10000 but received 6144
Expected to remove 10000 but removed 6912
    ✓ #4140 - HANGING (8633ms)


  1 passing (9s)
```

```
Received transfer when credit was 0
Expected 10000 but received 4346
Expected to remove 10000 but removed 4602
    ✓ #4140 - HANGING (8328ms)


  1 passing (8s)
```